temporalio/worker/_nexus.py

```
@@ -0,0 +1,475 @@
+                    callback_headers=dict(start_request.callback_header),
+                )
+                input = nexusrpc.handler.LazyValue(
+                    serializer=_DummyPayloadSerializer(
+                        data_converter=self._data_converter,
+                        payload=start_request.payload,
+                    ),
+                    headers={},
+                    stream=None,
+                )
+                try:
+                    result = await self._handler.start_operation(ctx, input)
+                except (
+                    nexusrpc.handler.UnknownServiceError,
+                    nexusrpc.handler.UnknownOperationError,
+                ) as err:
+                    # TODO(nexus-prerelease): error message
+                    raise nexusrpc.handler.HandlerError(
+                        "No matching operation handler",
+                        type=nexusrpc.handler.HandlerErrorType.NOT_FOUND,
+                        cause=err,
+                        retryable=False,
+                    ) from err
+
+            except nexusrpc.handler.OperationError as err:
+                return temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                    task_token=task_token,
+                    completed=temporalio.api.nexus.v1.Response(
+                        start_operation=temporalio.api.nexus.v1.StartOperationResponse(
+                            operation_error=await self._operation_error_to_proto(err),
+                        ),
+                    ),
+                )
+            except BaseException as err:
+                handler_err = _exception_to_handler_error(err)
+                return temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                    task_token=task_token,
+                    error=temporalio.api.nexus.v1.HandlerError(
+                        error_type=handler_err.type.value,
+                        failure=await self._exception_to_failure_proto(
```

Failure conversion exceptions do occur and it would be swallowed in this case

---

Thanks. Changed to
```python
        try:
            api_failure = temporalio.api.failure.v1.Failure()
            await self._data_converter.encode_failure(err, api_failure)
            api_failure = google.protobuf.json_format.MessageToDict(api_failure)
            return temporalio.api.nexus.v1.Failure(
                message=api_failure.pop("message", ""),
                metadata={"type": "temporal.api.failure.v1.Failure"},
                details=json.dumps(api_failure).encode("utf-8"),
            )
        except Exception as err:
            return temporalio.api.nexus.v1.Failure(
                message=f"{err.__class__.__name__}: {err}",
                metadata={"type": "temporal.api.failure.v1.Failure"},
            )
```